riscv-asm.md

```diff
@@ -384,6 +494,97 @@ as seen by `objdump`:
 			1c: R_RISCV_PCREL_LO12_S	.L0
 ```
 
+When used with the `%gprel` relocation function,
+the load and store pseudo instructions
+are used to load symbol addresses
+relative to `\_\_global_pointer$`,
+assumed to be in the `gp` register:
+
+```assembly
+	lw	a0, %gprel(var1)
+	fld	fa0, %gprel(var2), t0
+	sw	a0, %gprel(var3), t0
+	fsd	fa0, %gprel(var4), t0
+```
+
+Which generates the following assembler output and relocations
+as seen by `objdump`:
+
+```assembly
+0000000000000000 <.text>:
+   0:	00000537          	lui	a0,0x0
+			0: R_RISCV_GPREL_HI20	var1
+   4:	00350533          	add	a0,gp,a0
+			4: R_RISCV_GPREL_ADD	var1
+   8:	00052503          	lw	a0,0(a0)
+			8: R_RISCV_GPREL_LO12_I	var1
+   c:	000002b7          	lui	t0,0x0
+			c: R_RISCV_GPREL_HI20	var2
+  10:	003282b3          	add	t0,gp,t0
+			10: R_RISCV_GPREL_ADD	var2
+  14:	0002b507          	fld	fa0,0(t0)
+			14: R_RISCV_GPREL_LO12_I	var2
+  18:	000002b7          	lui	t0,0x0
+			18: R_RISCV_GPREL_HI20	var3
+  1c:	003282b3          	add	t0,gp,t0
+			1c: R_RISCV_GPREL_ADD	var3
+  20:	00a2a023          	sw	a0,0(t0)
+			20: R_RISCV_GPREL_LO12_S	var3
+  24:	000002b7          	lui	t0,0x0
+			24: R_RISCV_GPREL_HI20	var4
+  28:	003282b3          	add	t0,gp,t0
+			28: R_RISCV_GPREL_ADD	var4
+  1c:	00a2b027          	fsd	fa0,0(t0)
+			1c: R_RISCV_GPREL_LO12_S	var4
+```
+
+Thread Local Storage
+--------------
+
+The followig pseudo instruction is used for the TLS initial executable,
+relative to `\_\_global_pointer$`,
+assumed to be specified in the last register
+or defaulting to the `gp` register:
+
+```assembly
+	la.tls.ie.gprel	a0, msg + 1
+```
+
+Which generates the following assembler output and relocations,
+as seen by `objdump`:
+
+```assembly
+0000000000000000 <.text>:
+   0:	00000537          	lui	a0,0x0
+			0: R_RISCV_TLS_GOT_GPREL_HI20	msg+0x1
+   4:	00350533          	add	a0,gp,a0
+			4: R_RISCV_TLS_GOT_GPREL_ADD	msg+0x1
+   8:	00150513          	addi	a0,a0,1 # 0x1
+			8: R_RISCV_TLS_GOT_GPREL_LO12	msg+0x1
+```
+
+The followig pseudo instruction is used for the TLS global dynamic,
+relative to `\_\_global_pointer$`,
+assumed to be specified in the last register
+or defaulting to the `gp` register:
+
+```assembly
+	la.tls.gd.gprel	a0, msg + 1, s0
```

Comment:
Why does this one get a new mnemonic when the others use `%(got_)gprel` on the expression?

Reply:
Though there are the relocation functions `%tls_{gd,ie}_pcrel_hi`, there are also the pseudo instructions `la.tls.{gd,ie}`, which are implictiy PC relative.  This pseudo instruction is equivalent to the latter, but explicitly relative to the `__global_pointer$`.